File: 2. Matrix Transpose.cpp
// Topic: Matrix Transpose
#include <bits/stdc++.h>
using namespace std;
int main()
{
    vector<vector<int>> A = {{1, 2, 3}, {4, 5, 6}};  // 2x3
    int n = A.size(), m = A[0].size();
    vector<vector<int>> T(m, vector<int>(n));
    for (int i = 0; i < n; ++i)
        for (int j = 0; j < m; ++j) T[j][i] = A[i][j];
    cout << "Transpose:\n";
    for (auto& r : T)
    {
        for (auto v : r) cout << v << " ";
        cout << "\n";
    }
}

// TC: O(n*m)
// SC: O(n*m)

File: 2. Matrix Transpose.py
"""Topic: Matrix Transpose"""
def transpose(A):
    return list(map(list, zip(*A)))

if __name__=='__main__':
    A=[[1,2,3],[4,5,6]]
    print('Transpose:')
    for r in transpose(A): print(' '.join(map(str,r)))

# TC: O(n*m)
# SC: O(n*m)
